src/field/base/OdeArray.hh

```diff
@@ -0,0 +1,197 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file OdePointers.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Array.hh"
+#include "base/Range.hh"
+#include "base/Types.hh"
+
+#include <cmath>
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * A state array of the equation of motion: pos = v_[0-2], mom = v_[3-5]
+ *
+ * XXX TODO: extend this to include time and spin or to template with a
+ * variable ode_dim and array type
+ */
+
+class OdeArray
+{
+    static constexpr size_type ode_dim = 6;
+    using ode_type                     = Array<real_type, ode_dim>;
+
+  private:
+    ode_type v_;
+
+  public:
+    // Constructors
+    CELER_FUNCTION
+    OdeArray()
+    {
+        for (auto i : range(ode_dim))
+            v_[i] = 0;
+    }
+
+    CELER_FUNCTION
+    OdeArray(const real_type c)
+    {
+        for (auto i : range(ode_dim))
+            v_[i] = c;
+    }
+
+    CELER_FUNCTION
+    OdeArray(const ode_type v)
```

Comment:
Does the ODE array ever get constructed from an ODE type? Wouldn't it be better to construct from `const Real3& position, const Real3& momentum`? Did you consider using that for storage under the hood as well? It could make some of your operations easier as well as allowing const-reference access to the data as opposed to copy-by-value.

Reply:
I used this constructor in field tests (to initialize field states).  I was thinking a single interface to initialize the state when ode_type is extended later including time, spin and so on.  Nevertheless, this constructor will be never used as
the field state will be initialized based on geo/particle track view in a real workflow.  Anyway, the suggested one is better in the current ontext (6 variables with position and momentum), but the constructor will be dropped as there is another way to initialize the state in test routines.